pkg/daemon/update_test.go

```diff
@@ -20,6 +24,49 @@ import (
 	k8sfake "k8s.io/client-go/kubernetes/fake"
 )
 
+func newMockDaemon() Daemon {
+	// testClient is the NodeUpdaterClient mock instance that will front
+	// calls to update the host.
+	testClient := RpmOstreeClientMock{
+		GetBootedOSImageURLReturns: []GetBootedOSImageURLReturn{},
+	}
+
+	// Create a Daemon instance with mocked clients
+	return Daemon{
+		mock:              true,
+		name:              "nodeName",
+		os:                OperatingSystem{},
+		NodeUpdaterClient: testClient,
+		kubeClient:        k8sfake.NewSimpleClientset(),
+		bootedOSImageURL:  "test",
+	}
+}
+
+func setupTempDirWithEtc(t *testing.T) (string, func()) {
+	// Make a temp dir to put the testing files in, and make sure we clean it up
+	testDir, err := ioutil.TempDir("", "mco-test")
+	assert.Nil(t, err)
+
+	// Stub out a test directory structure -- we need to create /etc so createOrigFile can use it
+	etcDir := filepath.Join(testDir, "etc")
+	err = os.MkdirAll(etcDir, 0755)
+	assert.Nil(t, err)
```

Comment:
For these `assert.Nil` calls, use `require.Nil` instead. The reason for this is because calling `require.Nil` will cause `t.FailNow` to be called which will stop the test immediately. This is desirable here because this is setup / teardown code which should be fatal to the test. By comparison, if the code under test should return a `nil` value, use `assert.Nil` because it will allow the remainder of the test to continue.